File: internal/server/player.go
package server

import "github.com/lemunozm/ascii-arena/internal/pkg/comm"

type Player struct {
	character  byte
	connection *comm.Connection
	points     int
}

func NewPlayer(character uint8, connection *comm.Connection) *Player {
	return &Player{
		character:  character,
		connection: connection,
		points:     0,
	}
}

func (p Player) GetCharacter() uint8 {
	return p.character
}

func (p Player) GetConnection() *comm.Connection {
	return p.connection
}

func (p Player) GetPoints() int {
	return p.points
}

func (p *Player) AddPoints(points int) int {
	p.points += points
	return p.points
}

func (p *Player) ResetPoints() {
	p.points = 0
}

File: src/com/asciiarena/lib/server/player/Player.java
package com.asciiarena.lib.server.player;

import com.asciiarena.lib.common.communication.Connection;
import com.asciiarena.lib.common.logging.Log;

public class Player
{
    private char character;
    private Connection connection;
    private boolean connected;
    private int points;

    public Player(char character, Connection connection)
    {
        this.character = character;
        this.connection = connection;
        this.connected = true;
    }

    public void addPoints(int points)
    {
        this.points += points; 
    }

    public void markAsDisconnected()
    {
        Log.warning("Player %c disconnected", character);
        connected = false;
    }

    public boolean isConnected()
    {
        return connected;
    }

    public char getCharacter()
    {
        return character;
    }

    public Connection getConnection()
    {
        return connection;
    }

    public int getPoints()
    {
        return points;
    }
}
